src/Core/Tools/Enums/ReferenceEventSource.cs

```diff
@@ -10,4 +10,6 @@ public enum ReferenceEventSource
     User,
     [EnumMember(Value = "provider")]
     Provider,
+    [EnumMember(Value = "registrationStart")]
+    RegistrationStart,
```

Comment:
⛏️ I might call this "registration" instead of "registrationStart". That way it generalizes to any other kind of registration event we might want to track in the future.

Reply:
I wrestled with this a bit as well. I thought about using "registration", but I realized that for the registration finish endpoint we will be able to attach the reference event to a user after creation. So, it's unlikely for us to have other registration events.

Reply:
I think the source would still be the registration in that case. It involves a user, but not the same way as, say, creating a send or collection.

Reply:
I've been asked per Product on [the task](https://bitwarden.atlassian.net/browse/PM-7322) to ensure that the existing `ReferenceEventType.Signup` is preserved for the new flow:

```C#
    await _referenceEventService.RaiseEventAsync(
                            new ReferenceEvent(ReferenceEventType.Signup, user, _currentContext)
                            {
                                SignupInitiationPath = initiationPath
                            });
```

Reply:
You do you. The PR is approved.